### PR TITLE
decoding/encoding errors with brainvision .vhdr/.vmrk

### PR DESCRIPTION
**Pull request by dominikwelke**
solves #154 

note that my implementation adds a dependency to the chardet module
any thoughts on this?

I see 2 alternatives: 
- hardcoding `encoding='utf-8'`, if this would be brainvision-default (i'm not sure).
- extracting the encoding from the `Codepage='...'` line in the header, which would fall if there are files without this information.

### CODE REVIEW

**Comment by dominikwelke** [comment]
the latest commit works without the dependency.

it extracts decoding from `Codepage=...` entry in the .vhdr
if there is no such entry, it defaults to utf8

**Comment by jasmainak** [comment]
It's in CONTRIBUTING.md instead of README.md as that's what github recommends

File: mne_bids/utils.py [hunk 1]
```diff
@@ -487,6 +498,16 @@ def copyfile_brainvision(vhdr_src, vhdr_dest):
...
 
+    # extract encoding from brainvision header file, or default to utf-8
+    with open(vhdr_src, 'rb') as ef:
+        enc = ef.read()
+        if enc.find(b'Codepage=') != -1:
+            enc = enc[enc.find(b'Codepage=')+9:]
+            enc = enc.split()[0]
+            enc = enc.decode()
+        else:
+            enc = 'utf-8'
```
Comment (jasmainak) [inline mne_bids/utils.py]: there is code duplication. It could be more DRY with a private function

**Comment by jasmainak** [comment]
@dominikwelke can you also update whats_new page? Thanks

**Comment by dominikwelke** [comment]
@jasmainak - next commit will have the private function, and Ill update whats_new

@sappelhoff - yes, both versions (with/out dependency) solve my issue. atm I simply import my local (modified) git-fork but will implement the requested test in the next days. 

@jasmainak @sappelhoff - does mne-bids have some specific logging that i didn see, or just the print() commands? I'd add a log of the encoding + source (from header vs. default) to the next commit..

**Comment by sappelhoff** [comment]
No specific logging module, your plan sounds good. Looking forward to the test and private function.

File: doc/whats_new.rst [hunk 2]
```diff
@@ -18,6 +18,7 @@ Changelog
 Bug
 ~~~
 
+- Encoding of BrainVision .vhdr/.vmrk files is checked to prevent encoding/decoding errors when modifying, by Dominik Welke (`#155 <https://github.com/mne-tools/mne-bids/pull/155>`_)
```
Comment (sappelhoff) [inline doc/whats_new.rst]: It'd be good if you make your name a link to your github user page / or your personal webpage - see the last part of `whats_new.rst`

File: mne_bids/tests/test_utils.py [hunk 3]
```diff
@@ -113,6 +113,16 @@ def test_age_on_date():
         _age_on_date(bday, exp4)
 
 
+def test_get_brainvision_encoding():
```
[the remark this answers is not in the document]
Comment (dominikwelke) [reply mne_bids/tests/test_utils.py]: could you explain a bit more? - its the first test I write :)

[the remark this answers is not in the document]
Comment (sappelhoff) [reply mne_bids/tests/test_utils.py]: I am at a loss as well :-D +1 for an explanation @jasmainak - which regression? We just had a bug report and now we have a new feature covering that bug. The test verifies that the feature works as expected.

[the remark this answers is not in the document]
Comment (jasmainak) [reply mne_bids/tests/test_utils.py]: since you had a bug which we are trying to fix, the first step would be to reproduce the bug in a test. Otherwise, in the future, if someone changes something that reintroduces the bug, the test will not fail. Basically, I am asking -- can you test the corner case that you hit instead of testing for UTF-8 ?

[the remark this answers is not in the document]
Comment (dominikwelke) [reply mne_bids/tests/test_utils.py]: but I see your point. Ill check what I can do (python 3 apparently doesnt want users to change global decoding)

**Comment by jasmainak** [comment]
looks good to me now. Thanks @dominikwelke ! hope to see more contributions in the future :)

**Comment by jasmainak** [comment]
ps: you might want to link github email id with the email you use for your commits, so people can click on your name and go to your profile ...
